plugins/out_azure_kusto/azure_kusto_conf.c

```diff
@@ -0,0 +1,643 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*  Fluent Bit
+ *  ==========
+ *  Copyright (C) 2015-2022 The Fluent Bit Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <fluent-bit/flb_info.h>
+#include <fluent-bit/flb_jsmn.h>
+#include <fluent-bit/flb_oauth2.h>
+#include <fluent-bit/flb_output.h>
+#include <fluent-bit/flb_output_plugin.h>
+#include <fluent-bit/flb_sds.h>
+#include <fluent-bit/flb_time.h>
+#include <fluent-bit/flb_unescape.h>
+#include <fluent-bit/flb_upstream_ha.h>
+#include <fluent-bit/flb_utils.h>
+
+#include "azure_kusto.h"
+#include "azure_kusto_conf.h"
+
+static struct flb_upstream_node *flb_upstream_node_create_url(struct flb_azure_kusto *ctx,
```

Comment:
Oh I see, the thing is I took inspiration from other plugins and the examples in the contribution guide, which are written this way, so I tried to fit that coding style.

But I do agree it's more straight forward and error proof, I'll try to do some rewriting if I find the time :)

Reply:
Yes, I understand that it wasn't clearly stated anywhere else and I couldn't expect you to comply with what wasn't properly required because that wouldn't be fair. I just shared it to give you an example in case you had the time and wanted to do it but as I said I don't expect you to re-do it since the issue was on our side =)

Reply:
refactored it a bit to fit this style. Tried to also avoid too much of block nesting, but it might still need some improving, tell me what you think